agent/engine/task_manager_test.go

```diff
@@ -2211,19 +2211,94 @@ func TestTaskWaitForHostResources(t *testing.T) {
 	}()
 	time.Sleep(500 * time.Millisecond)
 
-	// Verify waiting queue is waiting at arn2
-	topTask, err := taskEngine.topTask()
+	// Verify waiting queue has arn2 only
+	task, err := taskEngine.getTaskByIndex(0)
+	assert.NoError(t, err)
+	assert.Equal(t, "arn2", task.Arn)
+	assert.Equal(t, 1, len(taskEngine.waitingTaskQueue))
+
+	// Remove task arn0
+	taskResources := taskEngine.managedTasks["arn0"].ToHostResources()
+	err = taskEngine.hostResourceManager.release("arn0", taskResources)
+	assert.NoError(t, err)
+	taskEngine.wakeUpTaskQueueMonitor()
+
+	time.Sleep(500 * time.Millisecond)
+
+	// Verify arn2 got dequeued and the queue is empty
+	_, err = taskEngine.getTaskByIndex(0)
+	assert.Error(t, err)
+	assert.Empty(t, taskEngine.waitingTaskQueue)
+}
+
+// TestTaskWaitForHostResourcesWithDifferentRequests tests task queuing behavior based on available host resources
+// for tasks with different resource requests
```

Comment:
Does this test still verify FIFO order? If not, should we add another test to verify queue monitor is able to follow non FIFO requests, where a task added to the queue later in the queue can get started if a task earlier in the queue does not have enough host resources available yet

Reply:
good catch, I intended this to follow the example in the PR description. I will reorder the tasks to reflect that.

Reply:
updated